docs/sdks/typescript/quickstart.md

```diff
@@ -142,365 +149,513 @@ function App() {
+        .withUri('ws://localhost:3000')
+        .withModuleName('quickstart-chat')
+        .withToken(localStorage.getItem('auth_token') || '')
+        .onConnect(onConnect)
+        .onDisconnect(onDisconnect)
+        .onConnectError(onConnectError)
+        .build()
+    );
+  }, []);
+```
 
-  let messagesType: MessageType[] = messages.map(message => {
-    let sender_identity = User.findByIdentity(message.sender);
-    let display_name = sender_identity
-      ? userNameOrIdentity(sender_identity)
-      : 'unknown';
+Here we are configuring our SpacetimeDB connection by specifying the server URI, module name, and a few callbacks including the `onConnect` callback. When `onConnect` is called after connecting, we store the connection state, our `Identity`, and our SpacetimeDB credentials in our React state. If there is an error connecting, we print that error to the console as well.
 
-    return {
-      name: display_name,
-      message: message.text,
-    };
-  });
+We are also using `localStorage` to store our SpacetimeDB credentials. This way, we can reconnect to SpacetimeDB with the same `Identity` and token if we refresh the page. The first time we connect, we won't have any credentials stored, so we pass `undefined` to the `withToken` method, this will cause SpacetimeDB to generate new credentials for us.
 
-  setMessages(messagesType);
-}
+If you chose a different name for your module, replace `quickstart-chat` with that name, or republish your module as `quickstart-chat`.
 
-client.current.on('initialStateSync', () => {
-  setAllMessagesInOrder();
-  var user = User.findByIdentity(local_identity?.current?.toUint8Array()!);
-  setName(userNameOrIdentity(user!));
-});
-```
+In the `onConnect` function we are also subscribing to the `message` and `user` tables. When we subscribe, SpacetimeDB will run our subscription queries and store the result in a local "SDK client cache". This cache will be updated in real-time as the data in the table changes on the server. The `onApplied` callback is called after SpacetimeDB has synchronized our subscribed data with the client cache.
 
-### Message.onInsert callback - Update messages
+### Accessing the Data
 
-When we receive a new message, we'll update the messages section of the page. Keep in mind that we only want to do this for new messages, i.e. those inserted by a `send_message` reducer invocation. When the server is initializing our cache, we'll get a callback for each existing message, but we don't want to update the page for those. To that effect, our `onInsert` callback will check if its `ReducerEvent` argument is not `undefined`, and only update the `message` section in that case.
+Once SpacetimeDB is connected, we can easily access the data inside of the client cache using our connection. The `conn.db` field allows you to access all of the tables of your database. Those tables will contain all data which you have subscribed to in your subscription configuration.
```

Comment:
```suggestion
Once SpacetimeDB is connected, we can easily access the data in the client cache using our `DbConnection`. The `conn.db` field allows you to access all of the tables of your database. Those tables will contain all data requested by your subscription configuration.
```

Reply:
Just want to flag here that I edited what you wrote:`DbConnection` to be `DBConnection`. This reflects the actual type name that we currently generate. It's typical (although contested in the various style guides) to fully capitalize initialisms. e.g.

The AirBnB styleguide: https://github.com/airbnb/javascript?tab=readme-ov-file#naming--Acronyms-and-Initialisms
Things like `ReactDOM`: https://legacy.reactjs.org/docs/react-dom.html